sunpy/map/mapbase.py

```diff
@@ -1464,23 +1471,89 @@ def submap(self, bottom_left, top_right=None):
+        Scale:			 [2.402792 2.402792] arcsec / pix
+        Reference Pixel:	 [126.5 126.5] pix
+        Reference Coord:	 [3.22309951 1.38578135] arcsec
+        array([[565.81494, 585.0416 , 656.4552 , 670.18854],
+            [516.1865 , 555.7032 , 634.7365 , 661.90424],
+            [620.9256 , 620.9256 , 654.8825 , 596.6707 ],
+            [667.5083 , 560.52094, 651.22766, 530.28534]], dtype=float32)
+
+        >>> bottom_left_vector = SkyCoord([0 * u.arcsec, 10 * u.arcsec], [0 * u.arcsec, 10 * u.arcsec], frame='heliographic_stonyhurst')
+        >>> aia.submap(bottom_left_vector)   # doctest: +REMOTE_DATA
+        <sunpy.map.sources.sdo.AIAMap object at 0x7f91aece8be0>
+        SunPy Map
+        ---------
+        Observatory:		 SDO
+        Instrument:		 AIA 3
+        Detector:		 AIA
+        Measurement:		 171.0 Angstrom
+        Wavelength:		 171.0 Angstrom
+        Observation Date:	 2011-06-07 06:33:02
+        Exposure Time:		 0.234256 s
+        Dimension:		 [4. 5.] pix
+        Coordinate System:	 helioprojective
+        Scale:			 [2.402792 2.402792] arcsec / pix
+        Reference Pixel:	 [1.5 1.5] pix
+        Reference Coord:	 [3.22309951 1.38578135] arcsec
+        array([[213.9748 , 256.76974, 244.54262, 356.62466],
+            [223.74321, 258.0102 , 292.27716, 340.65408],
+            [219.53459, 242.31648, 308.5911 , 331.373  ],
+            [268.24377, 254.83157, 268.24377, 321.89252],
+            [249.99167, 265.14267, 274.61206, 240.5223 ]], dtype=float32)
+        """
+        if not isinstance(bottom_left, u.Quantity):
+            bottom_left, top_right = get_rectangle_coordinates(bottom_left,
+                                                               top_right=top_right,
+                                                               width=width,
+                                                               height=height)
 
             bottom_left = self.world_to_pixel(bottom_left)
             top_right = self.world_to_pixel(top_right)
 
-        elif not (isinstance(bottom_left, u.Quantity) and bottom_left.unit.is_equivalent(u.pix) and
-                  isinstance(top_right, u.Quantity) and top_right.unit.is_equivalent(u.pix)):
-            raise ValueError("Invalid input, bottom_left and top_right must either be SkyCoord or Quantity in pixels.")
+        elif (
+                # You haven't given top_right or both width and height
+                (top_right is None and (width is None or height is None)) or
+                # You have given top_right but you have also given width and height
+                (top_right is not None and (width is not None or height is not None)) or
+                # You haven't given top_right, width or height
+                (top_right is None and width is None and height is None)
+        ):
+            raise ValueError("If bottom_left is not a SkyCoord either top_right or "
+                             "both width and height must be specified.")
```

Comment:
```suggestion
        elif ([arg is not None for arg in (bottom_left, top_right, width, height)]
              not in [[True, True, False, False], [True, False, True, True]]):
            raise ValueError("If bottom_left is not a SkyCoord either top_right alone or "
                             "both width and height must be specified.")
```
I think this is a more readable way to check for the two allowed patterns for None